Python/datacube.py

```diff
@@ -0,0 +1,76 @@
+import glob
+
+import numpy as np
+import xarray as xr
+import netCDF4 as nc
+
+def read_data(data_path, dv_key_name):
+    """Load an ensemble of CanRCM4
+    models into a single datacube.
+    ------------------------------
+    Args:
+        data_path (Str): path containing CanRCM4 ensemble
+        dv_key_name (Str): name of the design value as 
+            in the netCDF file
+    Returns:
+        ds (xarray Dataset): data cube of assembled ensemble models 
+            into a single variable
+    """
+    if not isinstance(data_path, str):
+        raise TypeError("data_path must be a string, got {}"
+                        .format(type(data_path)))
+
+    if not isinstance(dv_key_name, str):
+        raise TypeError("dv_key_name must be a string, got {}"
+                        .format(type(dv_key_name)))
+
+    # Create a list of all files in PATH
+    nc_list = np.asarray(glob.glob(data_path+"*.nc"))
+
+    for path in nc_list:
+        if path.endswith('.nc') is False:
+            raise IOError('{} is not a supported file type. Data directory \
+                          must have only .nc files'.format(path))
+
+    # create an example NetCDF4 dataset
+    inst = nc.Dataset(nc_list[0], 'r')
```

Comment:
I think you could use a context manager here, that way you don't have to remember to close the dataset.

```
with nc.Dataset(nc_list[0], 'r') as dataset:
    ...
```